microprofile/metrics-smallrye/src/main/java/org/wildfly/extension/microprofile/metrics/MicroProfileMetricsSubsystemDefinition.java

```diff
@@ -79,8 +89,43 @@ protected MicroProfileMetricsSubsystemDefinition() {
                 .setCapabilities(METRICS_REGISTRATION_RUNTIME_CAPABILITY, HTTP_CONTEXT_CAPABILITY));
     }
 
+    @Override
+    public void registerAttributes(ManagementResourceRegistration resourceRegistration) {
+            super.registerAttributes(resourceRegistration);
+            resourceRegistration.registerReadWriteAttribute(EXPOSED_SUBSYSTEMS, null, new ExposedSubsystemWriterHandler());
+    }
+
     @Override
     public Collection<AttributeDefinition> getAttributes() {
-        return Arrays.asList(ATTRIBUTES);
+        return Arrays.asList(SECURITY_ENABLED);
+    }
+
+    private class ExposedSubsystemWriterHandler extends ReloadRequiredWriteAttributeHandler {
+            private ExposedSubsystemWriterHandler() {
+                super(EXPOSED_SUBSYSTEMS);
+            }
+
+            @Override
+            protected void validateUpdatedModel(OperationContext context, Resource model) throws OperationFailedException {
+                super.validateUpdatedModel(context, model);
+                validExposedSubsystems(context, model.getModel());
```

Comment:
I'm not sure this works.

When the validation is called on the write-attribute op, the value of the attribute is in the `value` property.
However the `validExposedSubsystem` will read from the `exposed-subsystems` property (which is correct for the `:add` operation.